### ISSUE TITLE
after importing a cluster, rancher server shows this error [Ready False 38 mins ago [Disconnected] Cluster agent is not connected]

### ISSUE THREAD

**Issue by wang-xiaowu** [rancher/rancher#36589]
**Rancher Server Setup**
- Rancher version: v2.6.3
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): k3s
- Proxy/Cert Details: [k3s's ca](https://docs.rancher.cn/docs/rancher2/installation/resources/advanced/self-signed-ssl/_index/)

**Information about the Cluster**
- Kubernetes version: v1.21.7+k3s1
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Imported

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Admin

   
**Describe the bug**

![image](https://user-images.githubusercontent.com/44340137/154927501-fbc2bc88-486d-4159-9366-7762647f5d70.png)

**To Reproduce**
 
```
A-Server
0. /etc/profile - export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
1. curl -sfL http://rancher-mirror.cnrancher.com/k3s/k3s-install.sh | INSTALL_K3S_MIRROR=cn INSTALL_K3S_VERSION=v1.21.7+k3s1 sh -s - server
2. cat /var/lib/rancher/k3s/server/node-token

B-Server
0. /etc/profile - export K3S_TOKEN,export K3S_DATASTORE_ENDPOINT, export K3S_URL="https://10.0.2.15:6443" export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
1. curl -sfL http://rancher-mirror.cnrancher.com/k3s/k3s-install.sh | INSTALL_K3S_MIRROR=cn INSTALL_K3S_VERSION=v1.21.7+k3s1 sh -s - server

A-Agent
1. /etc/profile - export K3S_TOKEN, export K3S_URL="https://10.0.2.15:6443" export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
2. curl -sfL http://rancher-mirror.cnrancher.com/k3s/k3s-install.sh | INSTALL_K3S_MIRROR=cn INSTALL_K3S_VERSION=v1.21.7+k3s1 sh -

Install Rancher
1. https://docs.rancher.cn/docs/rancher2/installation/resources/advanced/self-signed-ssl/_index/
2. sh create_self-signed-cert.sh --ssl-domain=rancher.k3s.cn \
--ssl-trusted-ip=192.168.56.100,10.0.2.15 --ssl-size=2048 --ssl-date=3650
3. kubectl create namespace cattle-system
4. kubectl -n cattle-system create secret generic tls-ca \
  --from-file=cacerts.pem=./cacerts.pem

cp cacerts.pem ca-additional.pem
kubectl -n cattle-system create secret generic tls-ca-additional --from-file=ca-additional.pem

kubectl -n cattle-system create secret tls tls-rancher-ingress \
  --cert=tls.crt \
  --key=tls.key
5. helm install rancher ./rancher \
 --namespace cattle-system \
 --set hostname=rancher.k3s.cn \
 --set replicas=-1 \
 --set ingress.tls.source=secret \
 --set additionalTrustedCAs=true \
 --set useBundledSystemChart=true \
 --set privateCA=true
```
**Result**
```
[root@k3s-node1 rancher]# kubectl get svc -n cattle-system
NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
cattle-cluster-agent   ClusterIP   10.43.243.246   <none>        80/TCP,443/TCP   38m
rancher                ClusterIP   10.43.239.164   <none>        80/TCP,443/TCP   58m
rancher-webhook        ClusterIP   10.43.74.101    <none>        443/TCP          56m
webhook-service        ClusterIP   10.43.70.214    <none>        443/TCP          56m
[root@k3s-node1 rancher]# kubectl get deployments -n cattle-system
NAME                   READY   UP-TO-DATE   AVAILABLE   AGE
cattle-cluster-agent   2/2     2            2           39m
rancher                1/1     1            1           58m
rancher-webhook        1/1     1            1           56m
[root@k3s-node1 rancher]# clear
[root@k3s-node1 rancher]# kubectl get pods -n cattle-system
NAME                                    READY   STATUS      RESTARTS   AGE
cattle-cluster-agent-7bcb4b8dd9-2th4m   1/1     Running     0          43m
cattle-cluster-agent-7bcb4b8dd9-s65kh   1/1     Running     0          43m
helm-operation-27g5m                    0/2     Completed   0          46m
rancher-584b98cd56-ftnlx                1/1     Running     1          66m
rancher-webhook-5d4f5b7f6d-285zn        1/1     Running     0          63m
[root@k3s-node1 rancher]# kubectl logs helm-operation-27g5m -n cattle-system
error: a container name must be specified for pod helm-operation-27g5m, choose one of: [helm proxy]
[root@k3s-node1 rancher]# kubectl logs helm-operation-27g5m -n cattle-system heml
error: container heml is not valid for pod helm-operation-27g5m
[root@k3s-node1 rancher]# kubectl logs helm-operation-27g5m -n cattle-system helm
helm upgrade --force-adopt=true --history-max=5 --install=true --namespace=cattle-fleet-system --reset-values=true --timeout=5m0s --values=/home/shell/helm/values-fleet-100.0.2-up0.3.8.yaml --version=100.0.2+up0.3.8 --wait=true fleet /home/shell/helm/fleet-100.0.2-up0.3.8.tgz
checking 15 resources for changes
Looks like there are no changes for ServiceAccount "gitjob"
Looks like there are no changes for ServiceAccount "fleet-controller"
Looks like there are no changes for ServiceAccount "fleet-controller-bootstrap"
Looks like there are no changes for ClusterRole "gitjob"
Looks like there are no changes for ClusterRole "fleet-controller"
Looks like there are no changes for ClusterRole "fleet-controller-bootstrap"
Looks like there are no changes for ClusterRoleBinding "gitjob-binding"
Looks like there are no changes for ClusterRoleBinding "fleet-controller"
Looks like there are no changes for ClusterRoleBinding "fleet-controller-bootstrap"
Looks like there are no changes for Role "fleet-controller"
Looks like there are no changes for RoleBinding "fleet-controller"
Looks like there are no changes for Service "gitjob"
Looks like there are no changes for Deployment "gitjob"
Looks like there are no changes for Deployment "fleet-controller"
beginning wait for 15 resources with timeout of 5m0s
Release "fleet" has been upgraded. Happy Helming!
NAME: fleet
LAST DEPLOYED: Mon Feb 21 08:36:56 2022
NAMESPACE: cattle-fleet-system
STATUS: deployed
REVISION: 2
TEST SUITE: None

---------------------------------------------------------------------
SUCCESS: helm upgrade --force-adopt=true --history-max=5 --install=true --namespace=cattle-fleet-system --reset-values=true --timeout=5m0s --values=/home/shell/helm/values-fleet-100.0.2-up0.3.8.yaml --version=100.0.2+up0.3.8 --wait=true fleet /home/shell/helm/fleet-100.0.2-up0.3.8.tgz
---------------------------------------------------------------------
[root@k3s-node1 rancher]# kubectl logs cattle-cluster-agent-7bcb4b8dd9-2th4m -n cattle-system
INFO: Environment: CATTLE_ADDRESS=10.42.0.19 CATTLE_CA_CHECKSUM=80753de35f3c48c44e6b1a15906e6a5d078f3d2913faab27f4a9f37a180fad7c CATTLE_CLUSTER=true CATTLE_CLUSTER_AGENT_PORT=tcp://10.43.243.246:80 CATTLE_CLUSTER_AGENT_PORT_443_TCP=tcp://10.43.243.246:443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_ADDR=10.43.243.246 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PORT=443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_PORT_80_TCP=tcp://10.43.243.246:80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_ADDR=10.43.243.246 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PORT=80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_SERVICE_HOST=10.43.243.246 CATTLE_CLUSTER_AGENT_SERVICE_PORT=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTP=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTPS_INTERNAL=443 CATTLE_CLUSTER_REGISTRY= CATTLE_FEATURES=embedded-cluster-api=false,fleet=false,monitoringv1=false,multi-cluster-management=false,multi-cluster-management-agent=true,provisioningv2=false,rke2=false CATTLE_INGRESS_IP_DOMAIN=sslip.io CATTLE_INSTALL_UUID=6bb0a1b9-1590-4914-be72-877a080c56ec CATTLE_INTERNAL_ADDRESS= CATTLE_IS_RKE=false CATTLE_K8S_MANAGED=true CATTLE_NODE_NAME=cattle-cluster-agent-7bcb4b8dd9-2th4m CATTLE_SERVER=https://rancher.k3s.cn CATTLE_SERVER_VERSION=v2.6.3
INFO: Using resolv.conf: search cattle-system.svc.cluster.local svc.cluster.local cluster.local nameserver 10.43.0.10 options ndots:5
INFO: https://rancher.k3s.cn/ping is accessible
INFO: rancher.k3s.cn resolves to 10.0.2.15
INFO: Value from https://rancher.k3s.cn/v3/settings/cacerts is an x509 certificate
time="2022-02-21T08:39:00Z" level=info msg="Listening on /tmp/log.sock"
time="2022-02-21T08:39:00Z" level=info msg="Rancher agent version v2.6.3 is starting"
time="2022-02-21T08:39:00Z" level=info msg="Connecting to wss://rancher.k3s.cn/v3/connect/register with token starting with gfk2xj2cw7hkm78c82xjqxs4cgk"
time="2022-02-21T08:39:00Z" level=info msg="Connecting to proxy" url="wss://rancher.k3s.cn/v3/connect/register"
time="2022-02-21T08:39:00Z" level=info msg="Starting /v1, Kind=Service controller"
[root@k3s-node1 rancher]# kubectl logs cattle-cluster-agent-7bcb4b8dd9-2th4m -n cattle-system^C
[root@k3s-node1 rancher]# kubectl get pods -n cattle-system
NAME                                    READY   STATUS      RESTARTS   AGE
cattle-cluster-agent-7bcb4b8dd9-2th4m   1/1     Running     0          45m
cattle-cluster-agent-7bcb4b8dd9-s65kh   1/1     Running     0          44m
helm-operation-27g5m                    0/2     Completed   0          47m
rancher-584b98cd56-ftnlx                1/1     Running     1          67m
rancher-webhook-5d4f5b7f6d-285zn        1/1     Running     0          65m
[root@k3s-node1 rancher]# kubectl logs cattle-cluster-agent-7bcb4b8dd9-s65kh -n cattle-system
INFO: Environment: CATTLE_ADDRESS=10.42.2.5 CATTLE_CA_CHECKSUM=80753de35f3c48c44e6b1a15906e6a5d078f3d2913faab27f4a9f37a180fad7c CATTLE_CLUSTER=true CATTLE_CLUSTER_AGENT_PORT=tcp://10.43.243.246:80 CATTLE_CLUSTER_AGENT_PORT_443_TCP=tcp://10.43.243.246:443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_ADDR=10.43.243.246 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PORT=443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_PORT_80_TCP=tcp://10.43.243.246:80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_ADDR=10.43.243.246 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PORT=80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_SERVICE_HOST=10.43.243.246 CATTLE_CLUSTER_AGENT_SERVICE_PORT=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTP=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTPS_INTERNAL=443 CATTLE_CLUSTER_REGISTRY= CATTLE_FEATURES=embedded-cluster-api=false,fleet=false,monitoringv1=false,multi-cluster-management=false,multi-cluster-management-agent=true,provisioningv2=false,rke2=false CATTLE_INGRESS_IP_DOMAIN=sslip.io CATTLE_INSTALL_UUID=6bb0a1b9-1590-4914-be72-877a080c56ec CATTLE_INTERNAL_ADDRESS= CATTLE_IS_RKE=false CATTLE_K8S_MANAGED=true CATTLE_NODE_NAME=cattle-cluster-agent-7bcb4b8dd9-s65kh CATTLE_SERVER=https://rancher.k3s.cn CATTLE_SERVER_VERSION=v2.6.3
INFO: Using resolv.conf: search cattle-system.svc.cluster.local svc.cluster.local cluster.local nameserver 10.43.0.10 options ndots:5
INFO: https://rancher.k3s.cn/ping is accessible
INFO: rancher.k3s.cn resolves to 10.0.2.15
INFO: Value from https://rancher.k3s.cn/v3/settings/cacerts is an x509 certificate
time="2022-02-21T08:39:19Z" level=info msg="Listening on /tmp/log.sock"
time="2022-02-21T08:39:19Z" level=info msg="Rancher agent version v2.6.3 is starting"
time="2022-02-21T08:39:19Z" level=info msg="Connecting to wss://rancher.k3s.cn/v3/connect/register with token starting with gfk2xj2cw7hkm78c82xjqxs4cgk"
time="2022-02-21T08:39:19Z" level=info msg="Connecting to proxy" url="wss://rancher.k3s.cn/v3/connect/register"
time="2022-02-21T08:39:19Z" level=info msg="Starting /v1, Kind=Service controller"
[root@k3s-node1 rancher]# kubectl get pods -n cattle-system
NAME                                    READY   STATUS      RESTARTS   AGE
cattle-cluster-agent-7bcb4b8dd9-2th4m   1/1     Running     0          45m
cattle-cluster-agent-7bcb4b8dd9-s65kh   1/1     Running     0          45m
helm-operation-27g5m                    0/2     Completed   0          48m
rancher-584b98cd56-ftnlx                1/1     Running     1          68m
rancher-webhook-5d4f5b7f6d-285zn        1/1     Running     0          66m
[root@k3s-node1 rancher]# kubectl get pods -n cattle-system
NAME                                    READY   STATUS      RESTARTS   AGE
cattle-cluster-agent-7bcb4b8dd9-2th4m   1/1     Running     0          48m
cattle-cluster-agent-7bcb4b8dd9-s65kh   1/1     Running     0          48m
helm-operation-27g5m                    0/2     Completed   0          51m
rancher-584b98cd56-ftnlx                1/1     Running     1          71m
rancher-webhook-5d4f5b7f6d-285zn        1/1     Running     0          69m
[root@k3s-node1 rancher]# kubectl get pods -n cattle-system
NAME                                    READY   STATUS      RESTARTS   AGE
cattle-cluster-agent-7bcb4b8dd9-2th4m   1/1     Running     0          54m
cattle-cluster-agent-7bcb4b8dd9-s65kh   1/1     Running     0          54m
helm-operation-27g5m                    0/2     Completed   0          57m
rancher-584b98cd56-ftnlx                1/1     Running     1          77m
rancher-webhook-5d4f5b7f6d-285zn        1/1     Running     0          74m
[root@k3s-node1 rancher]# ^C
[root@k3s-node1 rancher]# kubectl logs helm-operation-27g5m -n cattle-system
error: a container name must be specified for pod helm-operation-27g5m, choose one of: [helm proxy]
[root@k3s-node1 rancher]# kubectl logs helm-operation-27g5m -n cattle-system helm
helm upgrade --force-adopt=true --history-max=5 --install=true --namespace=cattle-fleet-system --reset-values=true --timeout=5m0s --values=/home/shell/helm/values-fleet-100.0.2-up0.3.8.yaml --version=100.0.2+up0.3.8 --wait=true fleet /home/shell/helm/fleet-100.0.2-up0.3.8.tgz
checking 15 resources for changes
Looks like there are no changes for ServiceAccount "gitjob"
Looks like there are no changes for ServiceAccount "fleet-controller"
Looks like there are no changes for ServiceAccount "fleet-controller-bootstrap"
Looks like there are no changes for ClusterRole "gitjob"
Looks like there are no changes for ClusterRole "fleet-controller"
Looks like there are no changes for ClusterRole "fleet-controller-bootstrap"
Looks like there are no changes for ClusterRoleBinding "gitjob-binding"
Looks like there are no changes for ClusterRoleBinding "fleet-controller"
Looks like there are no changes for ClusterRoleBinding "fleet-controller-bootstrap"
Looks like there are no changes for Role "fleet-controller"
Looks like there are no changes for RoleBinding "fleet-controller"
Looks like there are no changes for Service "gitjob"
Looks like there are no changes for Deployment "gitjob"
Looks like there are no changes for Deployment "fleet-controller"
beginning wait for 15 resources with timeout of 5m0s
Release "fleet" has been upgraded. Happy Helming!
NAME: fleet
LAST DEPLOYED: Mon Feb 21 08:36:56 2022
NAMESPACE: cattle-fleet-system
STATUS: deployed
REVISION: 2
TEST SUITE: None

---------------------------------------------------------------------
SUCCESS: helm upgrade --force-adopt=true --history-max=5 --install=true --namespace=cattle-fleet-system --reset-values=true --timeout=5m0s --values=/home/shell/helm/values-fleet-100.0.2-up0.3.8.yaml --version=100.0.2+up0.3.8 --wait=true fleet /home/shell/helm/fleet-100.0.2-up0.3.8.tgz
---------------------------------------------------------------------
[root@k3s-node1 rancher]# 
```
**Expected Result**

**Screenshots**

![image](https://user-images.githubusercontent.com/44340137/154927803-877863bc-2fa8-4d00-9574-756b6425a711.png)
![image](https://user-images.githubusercontent.com/44340137/154927933-f6dbf792-6834-4c98-b131-f42e19f2cd44.png)
![image](https://user-images.githubusercontent.com/44340137/154927934-85fd72aa-7f6a-41a5-b2bd-273da0db88e2.png)

**Additional context**

**Comment by kpanic9** [comment]
I'm having the same issue after upgrading from 2.5.9 to 2.6.3-patch1. EKS clusters already imported to rancher 2.5.9 have this issue after upgrade.

**Comment by kpanic9** [comment]
Seems like the same issue as  #36589

**Comment by wang-xiaowu** [comment]
> Seems like the same issue as #35976

hey man,have u resolved the problem?

**Comment by wang-xiaowu** [comment]
> Seems like the same issue as #35976

i can't

**Comment by kpanic9** [comment]
I couldn't figure out a fix for the issue. I ended up downgrading the rancher to 2.6.2, that version doesn't have this problem.

**Comment by wang-xiaowu** [comment]
> I couldn't figure out a fix for the issue. I ended up downgrading the rancher to 2.6.2, that version doesn't have this problem.

thank u ,i'll try that version
but i wish that someone could find and resolve this problem
(in fact,i do not know it's a bug or my mistake on operation)

**Comment by wang-xiaowu** [comment]
> > I couldn't figure out a fix for the issue. I ended up downgrading the rancher to 2.6.2, that version doesn't have this problem.
> 
> thank u ,i'll try that version but i wish that someone could find and resolve this problem (in fact,i do not know it's a bug or my mistake on operation)

yeah!!! it's done, thank u!!!

**Comment by tarekabouzeid** [comment]
Hi, 
we also have the same exact problem with rancher 2.6.3 when we provision a new cluster using RKE.

**Comment by wang-xiaowu** [comment]
> Hi, we also have the same exact problem with rancher 2.6.3 when we provision a new cluster using RKE.

so it's a consensus issue

**Comment by tarekabouzeid** [comment]
@wang-xiaowu are you also using proxy in your setup ?

**Comment by wang-xiaowu** [comment]
> @wang-xiaowu are you also using proxy in your setup ?

yes

**Comment by kpanic9** [comment]
And same as #32934. I'm seeing the same error logs on rancher container. Looks like rancher team is refusing to ack this as an actual problem.

**Comment by wang-xiaowu** [comment]
> And same as #32934. I'm seeing the same error logs on rancher container. Looks like rancher team is refusing to ack this as an actual problem.

yeah, and i have tried 2.6.2 and 2.5.12 both too
2.6.2 has no permission to click my cluster when i login as admin, it's grey
2.5.12 is ok

**Comment by Shaked** [comment]
Any update regarding this ticket? Currently facing the same with 2.6.3, `cattle-cluster-agent` is stuck: 

```
klf cattle-cluster-agent-7f8bbfb496-h2xf6
...
INFO: REDACTED resolves to REDACTED
time="2022-04-06T12:15:22Z" level=info msg="Listening on /tmp/log.sock"
time="2022-04-06T12:15:22Z" level=info msg="Rancher agent version v2.6.3 is starting"
time="2022-04-06T12:15:22Z" level=info msg="Connecting to wss://REDACTED/v3/connect/register with token starting with REDACTED"
time="2022-04-06T12:15:22Z" level=info msg="Connecting to proxy" url="wss://REDACTED/v3/connect/register"
time="2022-04-06T12:15:23Z" level=info msg="Starting /v1, Kind=Service controller"
<STUCK HERE>
```

**Comment by m2khosravi** [comment]
The problem is that [kubelet] cannot patch its node status sometimes, take look at the RKE cluster nodes which has status [Kubelet stopped posting node status] and restart docker services to solve this issue.

**Comment by wang-xiaowu** [comment]
> The problem is that [kubelet] cannot patch its node status sometimes, take look at the RKE cluster nodes which has status [Kubelet stopped posting node status] and restart docker services to solve this issue.

i used containerd and i have tried restart k3s many times

**Comment by haydursek** [comment]
Any update on this ? I'm having the same issue v2.6.5.

**Comment by mokhos** [comment]
Same issue here. any update?

**Comment by donnyyu** [comment]
Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem
`kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge`
This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

**Comment by dealboy** [comment]
> Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge` This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

this  fixed my case!

**Comment by wang-xiaowu** [comment]
> Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge` This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

i'll try later, thanks

**Comment by Seathex** [comment]
> Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge` This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

It fixes my issue too ! Thank you :)

**Comment by nestor-lobo-work** [comment]
You saved my day @donnyyu 👏 
I owe you a beer for sure🚀

**Comment by felipetarran** [comment]
How can i see my cluster ID ?

**Comment by jadjay** [comment]
> How can i see my cluster ID ?

`kubectl get clusters.management.cattle.io`

**Comment by wang-xiaowu** [comment]
> kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge

yes my man, it's working
and follow this, it's my all operations
ps: after configured the k3s and helm
```
0. helm repo add rancher-stable http://rancher-mirror.oss-cn-beijing.aliyuncs.com/server-charts/stable
1. kubectl create namespace cattle-system
2. kubectl -n cattle-system create secret tls tls-rancher-ingress \
  --cert=tls.crt \
  --key=tls.key
3. cp cacerts.pem ca-additional.pem
4. kubectl -n cattle-system create secret generic tls-ca-additional --from-file=ca-additional.pem=./ca-additional.pem
5. kubectl -n cattle-system create secret generic tls-ca \
--from-file=cacerts.pem=./cacerts.pem
6. helm install rancher rancher-stable/rancher \
 --version v2.6.3 \
 --namespace cattle-system \
 --set hostname=rancher.k3s.cn \
 --set bootstrapPassword=admin \
 --set ingress.tls.source=secret \
 --set privateCA=true \
 --set additionalTrustedCAs=true
7. kubectl -n cattle-system rollout status deploy/rancher
8. (because the cert was made by myself,so i need configured the hosts, otherwise it does not know my cert's domain name,unless you used ip instead of domain name)
kubectl -n cattle-system patch  deployments rancher --patch '{
    "spec": {
        "template": {
            "spec": {
                "hostAliases": [
                    {
                      "hostnames":
                      [
                        "rancher.k3s.cn"
                      ],
                      "ip": "192.168.56.100"
                    }
                ]
            }
        }
    }
}'
kubectl -n cattle-system patch  deployments cattle-cluster-agent --patch '{
    "spec": {
        "template": {
            "spec": {
                "hostAliases": [
                    {
                      "hostnames":
                      [
                        "rancher.k3s.cn"
                      ],
                      "ip": "192.168.56.100"
                    }
                ]
            }
        }
    }
}'
9. kubectl get clusters.management.cattle.io =><REPLACE_WITH_CLUSTERID>
10. kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge
11. it's done
```
![image](https://user-images.githubusercontent.com/44340137/178474761-4a1f08b4-a2fc-47b7-8b72-8c08d6ac9942.png)

**Comment by RizqyAlvindra** [comment]
this works for me thanks

> Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge` This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

**Comment by Mohitv1947** [comment]
> I couldn't figure out a fix for the issue. I ended up downgrading the rancher to 2.6.2, that version doesn't have this problem.

How to install particular rancher version what is the command?

**Comment by lptarik** [comment]
I am having same issue when i create anew rke cluster. I don’t know how to solve it

**Comment by wang-xiaowu** [comment]
> version what is the command?

use `--version`, like this, and dont forget to add helm chart repo

helm install rancher rancher-stable/rancher \
 --version v2.6.3 \
 --namespace cattle-system \
......

**Comment by wang-xiaowu** [comment]
> I am having same issue when i create anew rke cluster. I don’t know how to solve it

i think it is same with what i did in k3s, look at the 9 and 10 step in my comment

https://github.com/rancher/rancher/issues/36589#issuecomment-1181617212

**Comment by lptarik** [comment]
I am using 2.6.9 , so it is not possible to go back. Also tried 8 and 9. steps. same issue.
I see below logs in kubelet 


E1101 16:14:52.538235   10903 pod_workers.go:951] "Error syncing pod, skipping" err="failed to \"StartContainer\" for \"cluster-register\" with CrashLoopBackOff: \"back-off 2m40s restarting failed container=cluster-register pod=cattle-cluster-agent-94686c6db-p5bzd_cattle-system(02b98de8-d9bf-4b28-a03d-4f3ccc896be6)\"" pod="cattle-system/cattle-cluster-agent-94686c6db-p5bzd" podUID=02b98de8-d9bf-4b28-a03d-4f3ccc896be6

**Comment by wang-xiaowu** [comment]
> I am using 2.6.9 , so it is not possible to go back. Also tried 8 and 9. steps. same issue. I see below logs in kubelet
> 
> E1101 16:14:52.538235 10903 pod_workers.go:951] "Error syncing pod, skipping" err="failed to "StartContainer" for "cluster-register" with CrashLoopBackOff: "back-off 2m40s restarting failed container=cluster-register pod=cattle-cluster-agent-94686c6db-p5bzd_cattle-system(02b98de8-d9bf-4b28-a03d-4f3ccc896be6)"" pod="cattle-system/cattle-cluster-agent-94686c6db-p5bzd" podUID=02b98de8-d9bf-4b28-a03d-4f3ccc896be6

sorry, i am using 2.6.7, this version is ok
but i did not try 2.6.9

**Comment by miglocatiq** [comment]
I'm having the same problem with v2.6.9. After creating a cluster through the rancher API the cluster hangs in a waiting stage. I also tried 

> Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge` This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

but that didn't work.

**Comment by mickey-bob** [comment]
try to delete pod node-agent-cluster in namespace cattle-system, and delete all stopped container. It worked for me

**Comment by nRachid** [comment]
I am having the same problem with rancher 2.7.4, I tried `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge`
and also deleting the node-agent-cluster didn't work for me

**Comment by zahidhh** [comment]
i have resolved this issue in Rancher version v2.7.9 by editing deployments file of cattle-agent like ...

1) # kubectl edit deployment cattle-cluster-agent -n cattle-system

Set the following into its manifest file (search for dnsPolicy - it should be set to ClusterFirst; and insert the block there below that line). Please note that indentation is extremely important and must be just like in the snippet: 

hostAliases:
- hostnames:
  - ranche_fqdn
  ip: rancher_ip

2) Save the file. Agent should be now connected to Rancher and cluster should be visible in it.

Run this command force deployed the cattle agent
#kubectl replace --force -f /tmp/kubectl-edit-2334386749.yaml

**Comment by unseensenpai** [comment]
> Run [this](https://rancher.com/docs/rancher/v2.5/en/installation/resources/update-ca-cert/#method-1-kubectl-command) on rancher local cluster might fix your problem `kubectl patch clusters.management.cattle.io <REPLACE_WITH_CLUSTERID> -p '{"status":{"agentImage":"dummy"}}' --type merge` This command is used to trigger Rancher to redeploy the agent in the downstream cluster.

That works for me aswell. Like super fix.
